src/manifest.c

```diff
@@ -866,13 +900,30 @@ void link_manifests(struct manifest *m1, struct manifest *m2)
 			continue;
 		} /* else ret > 0  m1/file1 is after m2/file2 */
 		list2 = list2->next;
-		account_new_file();
+
+		/* Delta manifests don't include unchanged files in the FROM file list,
+		 * so files in the TO file list without a peer are only new when they
+		 * don't exist on the system. */
+		filename = mk_full_filename(path_prefix, file2->filename);
+		if (!verify_file_lazy(filename)) {
+			account_new_file();
+		}
+		free_string(&filename);
 	}
 
 	// Capture new files if they are at the tail end of the file list
 	while (list2) {
+		file2 = list2->data;
 		list2 = list2->next;
-		account_new_file();
+
+		/* Delta manifests don't include unchanged files in the FROM file list,
+		 * so files in the TO file list without a peer are only new when they
+		 * don't exist on the system. */
+		filename = mk_full_filename(path_prefix, file2->filename);
```

Comment:
Same question as above, but anyways at this point you are repeating code a lot. Make this its own function if you end up keeping it.